src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.html

```diff
@@ -534,25 +534,43 @@ <h4 *ngIf="isImport" id="jhi-programming-exercise-heading-import" jhiTranslate="
                 </label>
             </div>
         </div>
-        <div class="pt-4"></div>
-        <div class="form-group" *ngIf="programmingExercise.id && !isImport">
-            <label class="form-control-label" jhiTranslate="artemisApp.exercise.notificationText" for="field_notification_text">Notification Text</label>
-            <input minlength="3" type="text" class="form-control" name="notificationText" id="field_notification_text" [(ngModel)]="notificationText" />
+        <div class="update-programming-exercise-footer">
+            <div class="update-programming-exercise-footer-content">
+                <span *ngIf="isSaving" jhiTranslate="artemisApp.quizExercise.edit.saving" class="badge bg-secondary"> </span>
+                <ng-template #tooltipTranslate>
+                    <div *ngFor="let reason of getInvalidReasons()">
+                        <p
+                            *ngIf="(reason.translateValues | json) != '{}'"
+                            jhiTranslate="{{ reason.translateKey }}"
+                            [translateValues]="{ index: reason.translateValues.index, threshold: reason.translateValues.threshold }"
+                        ></p>
+                        <p *ngIf="(reason.translateValues | json) == '{}'" jhiTranslate="{{ reason.translateKey }}"></p>
+                    </div>
+                </ng-template>
+                <span
+                    *ngIf="editForm.form.invalid || !validIdeSelection() || !auxiliaryRepositoriesValid || submissionPolicyForm.invalid"
```

Comment:
Couldn't these checks be simplified to `getInvalidReasons().length > 0`? Since all of these checks should also result in an entry in the returned array.
The same would go for the disabled tag of the save button.

Reply:
Implemented the changes.